airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -842,16 +838,14 @@ def cleanup(self, pod: k8s.V1Pod, remote_pod: k8s.V1Pod):
         if self._killed or not remote_pod:
             return
 
-        istio_enabled = self.is_istio_enabled(remote_pod)
         pod_phase = remote_pod.status.phase if hasattr(remote_pod, "status") else None
 
         # if the pod fails or success, but we don't want to delete it
         if pod_phase != PodPhase.SUCCEEDED or self.on_finish_action == OnFinishAction.KEEP_POD:
             self.patch_already_checked(remote_pod, reraise=False)
 
-        failed = (pod_phase != PodPhase.SUCCEEDED and not istio_enabled) or (
-            istio_enabled and not container_is_succeeded(remote_pod, self.base_container_name)
-        )
+        # Consider pod failed if the pod has not succeeded and the base container within the pod has not succeeded
+        failed = pod_phase != PodPhase.SUCCEEDED and not container_is_succeeded(remote_pod, self.base_container_name)
```

Comment:
No because if either of the statements are `False` then `failed` should be `False`. It's the equivalent of `not (pod_phase == PodPhase.SUCCEEDED or container_is_succeeded(remote_pod, self.base_container_name))`.

Reply:
Looks like we are not handling the case when it has sidecars. sidecars are still running event after base container exists. We can't judge the task state based on the pod phase. Why can't we base container state alone?
`failed = not container_is_succeeded(remote_pod, self.base_container_name)`

Reply:
> Why can't we use base container state alone?

We could, but I was trying to maintain the existing behavior. I agree that base container state should be the only thing that determines the task state.

Reply:
Updated the behavior to only check base container state in 67f720188944ec4f59aac1d8c8704e21907196a8